static/js/browser2/statvar_hierarchy_search.tsx

```diff
@@ -0,0 +1,212 @@
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Component for rendering the search part of the stat var hierarchy including
+ * the search input and dropdown results.
+ */
+
+import React from "react";
+import axios from "axios";
+import _ from "lodash";
+import { StatVarGroupNodeType, StatVarNodeType } from "./util";
+
+interface StatVarHierarchySearchPropType {
+  statVarGroupsData: { [key: string]: StatVarGroupNodeType };
+  statVarsData: { [key: string]: StatVarNodeType };
+  onSelectionChange: (newSelection: string) => void;
+}
+
+interface StatVarHierarchySearchStateType {
+  query: string;
+  svgResults: string[];
+  svResults: string[];
+  showNoResultsMessage: boolean;
+}
+
+export class StatVarHierarchySearch extends React.Component<
+  StatVarHierarchySearchPropType,
+  StatVarHierarchySearchStateType
+> {
+  private delayTimer: NodeJS.Timeout;
+
+  constructor(props: StatVarHierarchySearchPropType) {
+    super(props);
+    this.state = {
+      query: "",
+      svgResults: [],
+      svResults: [],
+      showNoResultsMessage: false,
+    };
+    this.onInputChanged = this.onInputChanged.bind(this);
+    this.search = this.search.bind(this);
+    this.onResultSelected = this.onResultSelected.bind(this);
+  }
+
+  render(): JSX.Element {
+    const renderResults =
+      !_.isEmpty(this.state.svResults) ||
+      !_.isEmpty(this.state.svgResults) ||
+      this.state.showNoResultsMessage;
+    return (
+      <div className="statvar-hierarchy-search-section">
+        <input
+          className="statvar-search-input"
+          type="text"
+          value={this.state.query}
+          onChange={this.onInputChanged}
+          placeholder="Search"
+          onBlur={() => this.setState({ showNoResultsMessage: false })}
+        />
+        {renderResults ? (
```

Comment:
nit: just "&&" here if there is no else clause required

Reply:
done